nucypher/cli/commands/contacts.py

```diff
@@ -0,0 +1,116 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import shutil
+
+import click
+import os
+
+from nucypher.characters.control.emitters import StdoutEmitter
+from nucypher.cli.actions.select import select_card
+from nucypher.cli.options import option_force
+from nucypher.cli.painting.policies import paint_single_card, paint_cards
+from nucypher.cli.types import EXISTING_READABLE_FILE
+from nucypher.policy.identity import Card
+
+
+@click.group()
+def contacts():
+    """"Manage character cards"""
+
+
+@contacts.command()
+@click.option('--id', 'card_identifier', type=click.STRING, required=False)
+@click.option('--name', 'card_nickname', type=click.STRING, required=False)
+@click.option('--qrcode', is_flag=True, default=None)
+def lookup(card_identifier, card_nickname, qrcode):
+    """"Manage character cards"""
+    emitter = StdoutEmitter()
+    if card_nickname and card_identifier:
+
+        raise click.Abort
+    card = select_card(emitter=emitter, card_identifier=card_identifier or card_nickname)
+    paint_single_card(emitter=emitter, card=card, qrcode=qrcode)
+
+
+@contacts.command()
+def all():
+    emitter = StdoutEmitter()
+    card_filepaths = os.listdir(Card.CARD_DIR)
+    if not card_filepaths:
+        emitter.error('No cards found.')
+    cards = list()
+    for filename in card_filepaths:
+        card_id, ext = filename.split('.')
+        card = Card.load(identifier=card_id)
+        cards.append(card)
+    paint_cards(emitter=emitter, cards=cards, as_table=True)
+
+
+@contacts.command()
+@click.option('--id', 'card_id', type=click.STRING, required=False)
+@click.option('--type', 'character_flag', type=click.STRING, required=False)
+@click.option('--verifying-key', type=click.STRING, required=False)
+@click.option('--encrypting-key', type=click.STRING, required=False)
+@click.option('--nickname', type=click.STRING, required=False)
+def create(emitter, card_id, character_flag, verifying_key, encrypting_key, nickname, force):
+    emitter = StdoutEmitter()
+
+    if not all((character_flag, verifying_key, encrypting_key)) and force:
```

Comment:
Seem to be having some issues with this line (after the edits to `emitter`, and `force`):

```bash
File "/Users/derek/Documents/Github/repos/forks/kieran/nucypher/nucypher/cli/commands/contacts.py", line 74, in create
    if not all((character_flag, verifying_key, encrypting_key)) and force:
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 781, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 700, in make_context
    self.parse_args(ctx, args)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 1045, in parse_args
    opts, args, param_order = parser.parse_args(args=args)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/parser.py", line 269, in parse_args
    self._process_args_for_options(state)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/parser.py", line 290, in _process_args_for_options
    arglen = len(arg)
TypeError: object of type 'NoneType' has no len()
```
I was running `nucypher contacts create` when I got the above, but I am able to create a card after commenting the line out - unsure what's going on there.

Reply:
:thinking: cannot reproduce - perhaps this was fixed by side-effect of RFCs.

Reply:
Probably stale - that comment was from my first round of review in August 😅